src/connector/src/sink/elasticsearch_opensearch/elasticsearch_opensearch_formatter.rs

```diff
@@ -152,7 +154,30 @@ impl ElasticSearchOpenSearchFormatter {
             match op {
                 Op::Insert | Op::UpdateInsert => {
                     let key = self.key_encoder.encode(rows)?.into_string()?;
-                    let value = self.value_encoder.encode(rows)?;
+                    let mut modified_col_indices = Vec::new();
+                    if let Some((delete_key, delete_row)) = update_delete_row.take()
+                        && delete_key == key
+                    {
+                        delete_row
+                            .iter()
+                            .enumerate()
+                            .zip_eq_debug(rows.iter())
+                            .for_each(|((index, delete_column), insert_column)| {
```

Comment:
Should we just compare the Option<delete_column> and Option<insert_column> directly? For a column that was None in delete_column and also None in insert_column, we also don't have to update the column.